src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Translator.java

```diff
@@ -355,4 +355,20 @@ public String translateTrue() {
     public String translateFalse() {
         return "FALSE";
     }
+
+    /**
+     * In H2 2.x, checking a boolean field against a boolean represented as a string causes an error.
+     * This code checks if the string is a boolean and processes it as a boolean, not as a string.
+     */
+    @Override
+    public String translate(final K k) {
+       if (k.getConstant() instanceof String) {
+            String s = (String) k.getConstant();
+            if (s.equalsIgnoreCase("TRUE") || s.equalsIgnoreCase("FALSE")) {
+                return Boolean.parseBoolean(s) ? translateTrue() : translateFalse();
+            }
+        }
```

Comment:
what if I have a string that happens to be "TRUE" but I really mean it as a string not a boolean??
instead of "TRUE", it's going to write in the DB "1"??

Reply:
The idea of this method is to consider "TRUE" or "FALSE" (strings) as TRUE or FALSE (booleans). Basically it removes the quotes. In a situation like this, the code would throw an error, because the unquoted value could not be stored in a string field.

I am aware that this implementation is not perfect. But CM is used to use booleans as strings in several select expressions, which is not allowed anymore in H2. So without this method, CM would have several breaking changes after the bump. So it is trade-off. In my opinion, it would be better to support that directly in PDB and from now on always use boolean comparisons as booleans, not strings.

Reply:
Ok, I see your point and I agree. I will remove that and clients of PDB will have to use boolean as boolean values.